codejail/jail_code.py

```diff
@@ -2,49 +2,30 @@
 
 # configure is a reexport for backwards compatibility
 from .jail import configure  # pylint: disable=unused-import
-from .jail import get_codejail, is_configured
+from .jail import get_codejail
 
 
 def jail_code(command, code=None, files=None, extra_files=None, argv=None,
               stdin=None, slug=None):
     """
     Run code in a jailed subprocess.
 
+    This function calls through to the `jail_code()` method on the `Jail`
+    object.
+
     `command` is an abstract command ("python", "node", ...) that must have
     been configured using `configure`.
 
-    `code` is a string containing the code to run.  If no code is supplied,
-    then the code to run must be in one of the `files` copied, and must be
-    named in the `argv` list.
-
-    `files` is a list of file paths, they are all copied to the jailed
-    directory.  Note that no check is made here that the files don't contain
-    sensitive information.  The caller must somehow determine whether to allow
-    the code access to the files.  Symlinks will be copied as symlinks.  If the
-    linked-to file is not accessible to the sandbox, the symlink will be
-    unreadable as well.
-
-    `extra_files` is a list of pairs, each pair is a filename and a bytestring
-    of contents to write into that file.  These files will be created in the
-    temp directory and cleaned up automatically.  No subdirectories are
-    supported in the filename.
-
-    `argv` is the command-line arguments to supply.
+    All other arguments, return values, and behavior are documented at
+    `codejail.jail.Jail.jail_code`
 
-    `stdin` is a string, the data to provide as the stdin for the process.
+    This function exists primarily for backwards compatibility.  For new code,
+    consider using `codejail.jail.Jail.jail_code` instead.
 
-    `slug` is an arbitrary string, a description that's meaningful to the
-    caller, that will be used in log messages.
-
-    Returns a JailResult object with:
-
-        .status: exit status of the process: an int, 0 for success
-        .stdout: stdout of the program, a string
-        .stderr: stderr of the program, a string
+    >>> import codejail.jail
+    >>> jail = codejail.jail.get_codejail("python")
+    >>> jail.jail_code(...)
     """
-    if not is_configured(command):
-        raise Exception("jail_code needs to be configured for %r" % command)
-
```

Comment:
Note: I'm removing this `Exception` because `get_codejail` (called next) already raises a `JailError` if the command is not configured.